README.md

```diff
@@ -605,13 +606,15 @@ Then, you will need to use the `e` option to manually choose which lines to add.
 <a href="unstaging-edits-and-staging-the-unstaged"></a>
 ### I want to stage my unstaged edits, and unstage my staged edits
 
-This is tricky. The best I figure is that you should stash your unstaged edits. Then, reset. After that, pop your stashed edits back, and add them.
+In many cases, You should unstage all files and then pick file you want and commit it.
+But assume that's what you want,You can create a temp commit to store your stage files, stage your unstaged files and stash them, Then, reset the temp commit and stash pop.
 
 ```sh
-$ git stash -k
-$ git reset --hard
-$ git stash pop
-$ git add -A
+$ git commit -m "WIP"
+$ git add . # also have untracked files
+$ git stash
+$ git reset HEAD^
+$ git stash pop --index 0
```

Comment:
What does `--index 0` do that `git stash pop` alone doesn't?

Reply:
I think it's a bug..not sure it's a bug of `magit` or `git` or my mind...I find that `git stash pop` make all stash unstaged. and `git stash pop --index 0` keep it staged.Maybe i could google it....

Reply:
Could you check?

Reply:
[here](https://stackoverflow.com/questions/31595873/git-stash-with-staged-files-does-stash-convert-staged-files-to-unstaged?answertab=active#tab-top)
I will add a note here.